libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -318,24 +343,24 @@ void AP_Camera::control(float session, float zoom_pos, float zoom_step, float fo
 /*
   Send camera feedback to the GCS
  */
-void AP_Camera::send_feedback(mavlink_channel_t chan) const
+void AP_Camera::send_feedback(mavlink_channel_t chan)
 {
     const AP_AHRS &ahrs = AP::ahrs();
-
+    update_offset_location();
```

Comment:
Ideally you'd actually only do this math once when we log the feedback, then use the same result for both the logging and for the sending of feedback, which would actually fix the bug where the MAVLink feedback is always time delayed and subtly wrong because of that.

Reply:
yep, I agree, but the current structure was using current_log and that would be updated between the call of the different camera function, so I didn't want to change much the behaviour. We could also get rid of the current_loc and get the data from the ahrs directly.